empathyco/x
# Set correctly the filters in History Query when selecting a Query Preview with filters

[EMP-2428](https://searchbroker.atlassian.net/browse/EMP-2428)

When we save the filters in the History Queries action update-history-queries-with-search-response.action.ts we were doing: `const filters = createHistoryQueriesFiltersList(searchResponse.request.filters);` In the case of filters that come from a `QueryPreview`, we only have` id` so we should compare the filters with the response to get the rest of the data.

In this PR we use facets with selected filters from the response to be able to provide a label for the `selectedFilters` although the stored item has only the `id` available.

**TEST**
Use VueDevTools and check the history queries state: 
* click a `queryPreview` with filters (e.g. `cortina`) and check the historyQuery's filters have a label.

Check that saving filters on history queries is still working properly:
* write a query & filter it, delete the query. If you select the query from history queries, the filters are applied.
* select a `queryPreview` with filters, and delete the query. If you select the query from history queries, the filters are applied.

[EMP-2427]: https://searchbroker.atlassian.net/browse/EMP-2427?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[EMP-2428]: https://searchbroker.atlassian.net/browse/EMP-2428?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
